src/algebra/jordan/special.lean

```diff
@@ -0,0 +1,277 @@
+@[simp] lemma unsym_zero [has_zero α] : unsym (0 : αˢʸᵐ) = 0 := rfl
+
+@[simp, to_additive] lemma sym_one [has_one α] : sym (1 : α) = 1 := rfl
+@[simp, to_additive] lemma unsym_one [has_one α] : unsym (1 : αˢʸᵐ) = 1 := rfl
+
+@[simp] lemma sym_add [has_add α] (a b : α) : sym (a + b) = sym a + sym b := rfl
+@[simp] lemma unsym_add [has_add α] (a b : αˢʸᵐ) : unsym (a + b) = unsym a + unsym b := rfl
+
+@[simp] lemma sym_neg [has_neg α] (a : α) : sym (-a) = -sym a := rfl
+@[simp] lemma unsym_neg [has_neg α] (a : αˢʸᵐ) : unsym (-a) = -unsym a := rfl
+
+@[simp, to_additive] lemma sym_inv [has_inv α] (a : α) : sym (a⁻¹) = (sym a)⁻¹ := rfl
+@[simp, to_additive] lemma unsym_inv [has_inv α] (a : αˢʸᵐ) : unsym (a⁻¹) = (unsym a)⁻¹ := rfl
+
+@[simp] lemma sym_sub [has_sub α] (a b : α) : sym (a - b) = sym a - sym b := rfl
+@[simp] lemma unsym_sub [has_sub α] (a b : αˢʸᵐ) : unsym (a - b) = unsym a - unsym b := rfl
+
+@[simp] lemma sym_smul {R : Type*} [has_scalar R α] (c : R) (a : α) : sym (c • a) = c • sym a := rfl
+@[simp] lemma unsym_smul {R : Type*} [has_scalar R α] (c : R) (a : αˢʸᵐ) :
+  unsym (c • a) = c • unsym a := rfl
+
+
+@[simp] lemma unsym_eq_zero_iff {α} [has_zero α] (a : αˢʸᵐ) : a.unsym = (0 : α) ↔ a = (0 : αˢʸᵐ) :=
+unsym_injective.eq_iff' rfl
+
+@[simp] lemma sym_eq_zero_iff [has_zero α] (a : α) : sym a = (0 : αˢʸᵐ) ↔ a = (0 : α) :=
+sym_injective.eq_iff' rfl
+
+lemma unsym_ne_zero_iff [has_zero α] (a : αˢʸᵐ) : a.unsym ≠ (0 : α) ↔ a ≠ (0 : αˢʸᵐ) :=
+not_congr $ unsym_eq_zero_iff a
+
+lemma sym_ne_zero_iff [has_zero α] (a : α) : sym a ≠ (0 : αˢʸᵐ) ↔ a ≠ (0 : α) :=
+not_congr $ sym_eq_zero_iff a
+
+@[simp] lemma unsym_eq_one_iff [has_one α] (a : αˢʸᵐ) : a.unsym = 1 ↔ a = 1 :=
+unsym_injective.eq_iff' rfl
+
+@[simp] lemma sym_eq_one_iff [has_one α] (a : α) : sym a = 1 ↔ a = 1 :=
+sym_injective.eq_iff' rfl
+
+instance [add_comm_semigroup α] : add_comm_semigroup (αˢʸᵐ) :=
+unsym_injective.add_comm_semigroup _ (λ _ _, rfl)
+
+instance [add_monoid α] : add_monoid (αˢʸᵐ) :=
+unsym_injective.add_monoid_smul _ rfl (λ _ _, rfl) (λ _ _, rfl)
+
+instance [add_group α] : add_group (αˢʸᵐ) :=
+unsym_injective.add_group_smul _ rfl
+  (λ _ _, rfl) (λ _, rfl) (λ _ _, rfl) (λ _ _, rfl) (λ _ _, rfl)
+
+instance [add_comm_monoid α] : add_comm_monoid (αˢʸᵐ) :=
+{ ..sym_alg.add_comm_semigroup, ..sym_alg.add_monoid }
+
+instance [add_comm_group α] : add_comm_group (αˢʸᵐ) :=
+{ ..sym_alg.add_comm_monoid, ..sym_alg.add_group }
+
+instance {R : Type*} [semiring R] [add_comm_monoid α] [module R α] : module R αˢʸᵐ :=
+function.injective.module R ⟨unsym, rfl, λ _ _, rfl⟩ (λ _ _, id) (λ _ _, rfl)
+
+lemma mul_def [ring α] [invertible (2 : α)] (a b : αˢʸᵐ) :
+  a * b = sym (⅟2*(unsym a * unsym b + unsym b * unsym a)) := by refl
+
+instance [has_mul α] [has_add α] [has_one α] [invertible (2 : α)] (a : α) [invertible a] :
+  invertible (sym a) :=
+{ inv_of := sym ⅟a,
+  inv_of_mul_self := begin
+    change ⅟2 * (⅟a * a + a * ⅟a) = 1,
```

Comment:
You should be using your `sym_mul_sym` lemma from below here, instead of `change`.

Reply:
Thanks. It doesn't seem to quite line up:
```
rewrite tactic failed, did not find instance of the pattern in the target expression
  @has_mul.mul.{u_1} (sym_alg.{u_1} α) (@sym_alg.has_mul.{u_1} α _inst_2 _inst_1 _inst_3 _inst_4)
    (@sym_alg.sym.{u_1} α (@invertible.inv_of.{u_1} α _inst_1 _inst_3 a _inst_5))
    (@sym_alg.sym.{u_1} α a)
state:
α : Type u_1,
_inst_1 : has_mul.{u_1} α,
_inst_2 : has_add.{u_1} α,
_inst_3 : has_one.{u_1} α,
_inst_4 : @invertible.{u_1} α _inst_1 _inst_3 (@bit0.{u_1} α _inst_2 (@has_one.one.{u_1} α _inst_3)),
a : α,
_inst_5 : @invertible.{u_1} α _inst_1 _inst_3 a
⊢ @eq.{u_1+1} (sym_alg.{u_1} α)
    (@has_mul.mul.{u_1} (sym_alg.{u_1} α) (@sym_alg.has_mul.{u_1} α _inst_2 _inst_1 _inst_3 _inst_4)
       (@sym_alg.sym.{u_1} (Π (a : α) [self : @invertible.{u_1} α _inst_1 _inst_3 a], α)
          (@invertible.inv_of.{u_1} α _inst_1 _inst_3)
          a
          _inst_5)
       (@sym_alg.sym.{u_1} α a))
    (@has_one.one.{u_1} (sym_alg.{u_1} α) (@sym_alg.has_one.{u_1} α _inst_3))
```

Reply:
`src/algebra/jordan/special.lean` has now been moved to #11399  and this issue resolved by https://github.com/leanprover-community/mathlib/pull/11399/commits/b52ab21c8f63b953a0f9eac05786a74f184a7175